Docs/src/ogre-shadows.md

```diff
@@ -397,15 +400,136 @@ Non-PCF block. Note that after doing this, the uSTexWidth and
-some pointer to a MovablePlane and a pointer to a light.
+## Debugging Shadows
+Since shadows are a difficult subject, so it is a good idea to have the Shadow Map 
+projected on a Mini-Screen where it is possible to see how the Depth Caster is performing.
+
+```cpp
+// Create vertex program definition
+auto vertexProgram = Ogre::HighLevelGpuProgramManager::getSingleton().createProgram(
+	"DebugVP", Ogre::ResourceGroupManager::INTERNAL_RESOURCE_GROUP_NAME,
+	"glsl", Ogre::GpuProgramType::GPT_VERTEX_PROGRAM);
+vertexProgram->setSourceFile("shadow_debug.vs");
+
+// Create fragment program definition
+auto fragmentProgram = Ogre::HighLevelGpuProgramManager::getSingleton().createProgram(
+	"DebugFP", Ogre::ResourceGroupManager::INTERNAL_RESOURCE_GROUP_NAME,
+	"glsl", Ogre::GpuProgramType::GPT_FRAGMENT_PROGRAM);
+fragmentProgram->setSourceFile("shadow_debug.fs");
+fragmentProgram->getDefaultParameters()->setNamedConstant("shadowMap", 0);
+
+// Create material to show the shadow map
+Ogre::MaterialPtr material = Ogre::MaterialManager::getSingleton().create(
+	"ShadowDebug", Ogre::ResourceGroupManager::INTERNAL_RESOURCE_GROUP_NAME);
+Ogre::Pass* firstPass = material->getTechnique(0)->getPass(0);
+firstPass->setVertexProgram("DebugVP");
+firstPass->setFragmentProgram("DebugFP");
+auto textureUnit = firstPass->createTextureUnitState();
+textureUnit->setTextureAddressingMode(Ogre::TAM_CLAMP);
+textureUnit->setTextureFiltering(Ogre::TFO_NONE);
+textureUnit->setContentType(Ogre::TextureUnitState::CONTENT_SHADOW);
+
+// Create rectangle for the Mini-Screen and attach to node
+Ogre::Rectangle2D* miniScreen = new Ogre::Rectangle2D(true);
+miniScreen->setCorners(.5, 1.0, 1.0, .5);
+miniScreen->setBoundingBox(Ogre::AxisAlignedBox::BOX_INFINITE);
+miniScreen->setMaterial(Ogre::MaterialManager::getSingletonPtr()->getByName("ShadowDebug"));
+
+Ogre::SceneNode* miniScreenNode = mSceneMgr->getRootSceneNode()->createChildSceneNode();
+miniScreenNode->attachObject(miniScreen);	
+```
+
+Vertex shader `shadow_debug.vs`:
+```glsl
+#version 330 core
+
+layout (location = 0) in vec4 vertex;
+layout (location = 8) in vec2 uv0;
+
+out vec2 TexCoords;
+
+void main()
+{
+    gl_Position = vec4(vertex.x, vertex.y, 0.0, 1.0); 
+    TexCoords = uv0;
+}
+```
+
+Fragment shader `shadow_debug.fs`:
+```glsl
+#version 330 core
+
+out vec4 FragColor;
+  
+in vec2 TexCoords;
+
+uniform sampler2D shadowMap;
+
+void main()
+{ 
+    FragColor = texture(shadowMap, TexCoords);
+}
+```
 
+## Improving Shadow Quality
+
+### Shadow Camera Setups {#sm_shadow_camera_setup}
+
+The default projection used when rendering shadow textures is a uniform frustum. 
+This is pretty straight forward but doesn't make the best use of the space in the shadow map 
+since texels closer to the camera will be larger, resulting in 'jaggies'. 
+There are several ways to distribute the texels in the shadow texture differently.
+Ogre is provided with several alternative shadow camera setups:
```

Comment:
there is some overlap with https://ogrecave.github.io/ogre/api/latest/_shadows.html#Texture_002dbased-Shadows

Reply:
True, but it is a small introduction to the code, perhaps both documents should be revised to see what should go where.

Also I wanted to ask if it is OK to add some Shader Code for the PSSM projection, it is based on the code that comes before but with modifications to make it work with PSSM...

Reply:
> Also I wanted to ask if it is OK to add some Shader Code for the PSSM projection

sure